### ISSUE TITLE
All-in-One `netrunner.json`

### ISSUE THREAD

**Issue by sboosali** [NetrunnerDB/netrunner-cards-json#853]
Useful for other applications to easily build their own search, completion, analysis, etc for Netrunner cards.

For example,

``` json
{
  "meta": {"version":…},

  "enums": {
     "types": {"ice":…},
     "flags": {"threat":…}
     "factions": …,
     …
  },

  "cards": {
    "…": {…},
    "…": {…},
    …
  },

  "sets": {
    "prints": [
       {…},
       {…}, 
       …
    ],
    …
  }

}
```

Compare with mtgjson's `AllPrintings.json.gz`: https://mtgjson.com/downloads/all-files/

**Comment by plural** [comment]
perhaps.

i think we are unlikely to prioritize doing a lot to directly spend time improving other apps.  NRDB's api is the search for netrunner and that is accessible via a public API.

**Comment by NoahTheDuke** [comment]
I generate something like this for jnet in noahtheduke/netrunner-data, in the file edn/raw_data.edn. it's helpful in some respects but it's pretty scouring to keep updated, i'll be real.

**Comment by sboosali** [comment]
@plural With respect to third party apps, my own usecase is just an Emacs mode I'm working on to edit custom cards, as well as to search for a few random predicates that aren't features of NRDB (like “ICE `sub_count`”, AFAICT). But yes, my plan was to just have an offline file that merges `/cards`/etc, and manually add any wanted fields that aren't available.

Obviously, I'll write some short script to concatenate this data myself (or just quasi-copy-paste @NoahTheDuke's lisp file, thanks!), but it sounds like you (@plural) wouldn't be interested in any pull requests for it? This would help other people writing, for example, a simple offline decklist editing mode for Emacs with completion and legality (which I don't currently intend to).

**Comment by plural** [comment]
@sboosali - if you want to contribute it, it will need good test coverage and you need to decide if you only want it to be v1 or v2 format.

If you want to contribute it, we will probably accept it.  We probably won't work on it ourselves.

**Comment by plural** [comment]
Additionally ,i would not want it to have any new types that only exist in that file.  If you want a mega-merge of all the types in one file, sure. If you wanted to turn card types into enums in a separate way like you described in your example, that seems like something specific to you/your personal need, but a needless divergence from the datatypes in the project.

**Comment by sboosali** [comment]
>If you wanted to turn card types into enums in a separate way like you described in your example,

Of course, it wasn't the intended schema, it was just an example, and from memory.

>If you want to contribute it, we will probably accept it. It will need good test coverage,

That sounds great. I'm asking because (while I'm happy to do any munging/testing/documenting/etc), I haven't contributed to Netrunner DB before, and different projects have different expectations.

Thanks for the quick replies.

**Comment by NoahTheDuke** [comment]
https://github.com/NoahTheDuke/netrunner-data/blob/master/edn/raw_data.edn

is the link to mine. it's not pretty printed because it's too big lol. it makes my neovim chug when i open it.
